### PR TITLE
fix: Fix the f16 and bf16 dump error & add all type test for csv & csv supports null

### PR DESCRIPTION
**Pull request by OxalisCu**
https://github.com/milvus-io/pymilvus/issues/2276

### CODE REVIEW

File: pymilvus/bulk_writer/buffer.py [hunk 1]
```diff
@@ -290,49 +289,62 @@ def _persist_parquet(self, local_path: str, **kwargs):
 
     def _persist_csv(self, local_path: str, **kwargs):
         sep = self._config.get("sep", ",")
+        # nullkey is not supported in csv now
```
Comment (tedxu) [inline pymilvus/bulk_writer/buffer.py:292]: Since CSV bulk insert has already supported nulls, can we add null support here?

File: pymilvus/bulk_writer/buffer.py [hunk 2]
```diff
@@ -290,49 +289,62 @@ def _persist_parquet(self, local_path: str, **kwargs):
...
+                data[k] = pd.Series(v, dtype=dt)
+            else:
+                data[k] = pd.Series(v)
 
         file_path = Path(local_path + ".csv")
         try:
-            with file_path.open("w", encoding="utf-8") as f:
-                writer = csv.writer(f, delimiter=sep)
-                writer.writerows(rows)
+            data.to_csv(file_path, sep=sep, index=False)
```
Comment (tedxu) [inline pymilvus/bulk_writer/buffer.py:345]: We should redefine the null value representation here, e.g., "NULL", configurable variable preferred.
